fluidsynth: CMake config broken

### Describe the bug
Adding a `dev` output to `fluidsynth` in #226253 (CC @trofi) broke its `export()`-generated CMake config.

```
CMake Error at /nix/store/m9w00irsx5f1lc0fqm0mgvl73ww9d1da-fluidsynth-2.3.2-dev/lib/cmake/fluidsynth/FluidSynthTargets.cmake:94 (message):
  The imported target "FluidSynth::libfluidsynth" references the file

     "/nix/store/m9w00irsx5f1lc0fqm0mgvl73ww9d1da-fluidsynth-2.3.2-dev/lib/libfluidsynth.so.3.2.0"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/nix/store/m9w00irsx5f1lc0fqm0mgvl73ww9d1da-fluidsynth-2.3.2-dev/lib/cmake/fluidsynth/FluidSynthTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /nix/store/m9w00irsx5f1lc0fqm0mgvl73ww9d1da-fluidsynth-2.3.2-dev/lib/cmake/fluidsynth/FluidSynthConfig.cmake:148 (include)
  builds/cmake/Modules/PlayerFindPackage.cmake:54 (find_package)
  CMakeLists.txt:1010 (player_find_package)
```

The generated CMake code in `FluidSynthTargets.cmake` determines an import prefix relative to the location of the the file.
```cmake
# Compute the installation prefix relative to this file.
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
if(_IMPORT_PREFIX STREQUAL "/")
  set(_IMPORT_PREFIX "")
endif()
```

And then proceeds to append the expected library location to that prefix in `FluidSynthTargets-release.cmake` when it prepares the corresponding CMake target.
```cmake
# Import target "FluidSynth::libfluidsynth" for configuration "Release"
set_property(TARGET FluidSynth::libfluidsynth APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
set_target_properties(FluidSynth::libfluidsynth PROPERTIES
  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libfluidsynth.so.3.2.0"
  IMPORTED_SONAME_RELEASE "libfluidsynth.so.3"
  )
```

### Steps To Reproduce
Steps to reproduce the behavior:
1. Build something that uses `fluidsynth`'s CMake config (I've noticed it while bumping `easyrpg-player` and trying to add `fluidsynth` to its `buildInputs`)
2. Observe error

### Expected behavior
The config should have a valid path and using it should work fine.

### Screenshots
n/a

### Additional context
n/a

### Notify maintainers

@cillianderoiste 
@lovek323 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.30-xanmod1, NixOS, 23.05 (Stoat), 23.05.419.3a70dd92993`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(bt1cn): `"unstable"`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/applications/audio/fluidsynth/default.nix
+++ b/pkgs/applications/audio/fluidsynth/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, lib, fetchFromGitHub, buildPackages, pkg-config, cmake
+{ stdenv, lib, fetchFromGitHub, fetchpatch, buildPackages, pkg-config, cmake
 , alsa-lib, glib, libjack2, libsndfile, libpulseaudio
 , AppKit, AudioUnit, CoreAudio, CoreMIDI, CoreServices
 }:
@@ -14,6 +14,16 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-BSJu3jB7b5G2ThXBUHUNnBGl55EXe3nIzdBdgfOWDSM=";
   };
 
+  patches = [
+    # Fixes bad CMAKE_INSTALL_PREFIX + CMAKE_INSTALL_LIBDIR concatenation for Darwin install name dir
+    # Remove when PR merged & in release
+    (fetchpatch {
+      name = "0001-Fix-incorrect-way-of-turning-CMAKE_INSTALL_LIBDIR-absolute.patch";
+      url = "https://github.com/FluidSynth/fluidsynth/pull/1261/commits/03cd38dd909fc24aa39553d869afbb4024416de8.patch";
+      hash = "sha256-nV+MbFttnbNBO4zWnPLpnnEuoiESkV9BGFlUS9tQQfk=";
+    })
+  ];
+
   outputs = [ "out" "dev" "man" ];
 
   nativeBuildInputs = [ buildPackages.stdenv.cc pkg-config cmake ];
@@ -24,8 +34,6 @@ stdenv.mkDerivation rec {
 
   cmakeFlags = [
     "-Denable-framework=off"
-    # set CMAKE_INSTALL_NAME_DIR to correct value on darwin
-    "-DCMAKE_INSTALL_LIBDIR=lib"
   ];
 
   meta = with lib; {

